README.md

```diff
@@ -44,6 +44,299 @@ This is an experimental framework that aims to combine the lessons learned from
 $ pip install pytorch-ie
 ```
 
+## 🥧 Concepts & Architecture
+
+PyTorch-IE builds on three core concepts: the **📃 Document**, the **🔤 ⇔ 🔢 Taskmodule**, and the **🧮 Model**. In a
+nutshell, the Document says how your data is structured, the Model defines your trainable logic and the Taskmodule
+converts from one end to the other. All three concepts are represented as abstract classes that should be used to
+derive use-case specific versions. In the following, they are explained in detail.
+
+<details>
+<summary>
+
+### 📃 Document
+
+</summary>
+
+The `Document` class is a special `dataclass` that defines the document model. Derivations can contain several
+elements:
+
+-   **Data fields** like strings to represent one or multiple texts or arrays for image data. These elements can be
+    arbitrary python objects.
+-   **Annotation fields** like labeled spans for entities or labeled tuples of spans for relations. These elements have
+    to be of a certain container type `AnnotationList` that is dynamically typed with the actual annotation type, e.g.
+    `entities: AnnotationList[LabeledSpan]`. Furthermore, annotation elements define one or multiple annotation `targets`.
+    An annotation target is either a data element or another annotation container. Internally, targets are used to construct the
+    annotation graph, i.e. data elements and annotation containers are the nodes and targets define the edges. The
+    annotation graph defines the (de-)serialization order and what is accessible from within an annotation. To
+    facilitate the setup of annotation containers, there is the `annotation_field()` method.
+-   **Other fields** to save metadata, ids, etc. They are not constrained in any way, but can not be accessed from within
+    annotations.
+
+<details>
+
+<summary>
+
+#### Example Document Model
+
+</summary>
+
+```python
+from typing import Optional
+from pytorch_ie.core import Document, AnnotationList, annotation_field
+from pytorch_ie.annotations import LabeledSpan, BinaryRelation, Label
+
+class MyDocument(Document):
+    # data fields (any field that is targeted by an annotation fields)
+    text: str
+    # annotation fields
+    entities: AnnotationList[LabeledSpan] = annotation_field(target="text")
+    relations: AnnotationList[BinaryRelation] = annotation_field(target="entities")
+    label: AnnotationList[Label] = annotation_field()
+    # other fields
+    doc_id: Optional[str] = None
+```
+
+Note that the `label` is a special annotation field that does not define a target because it belongs to the whole document.
+You can also have more complex constructs, like annotation fields that target multiple other fields by using
+`annotation_field(targets)` or `annotation_field(named_targets)`. The latter is useful if you want to access the
```

Comment:
to me, here it is unclear what the difference to the above example annotation_field(target="text") is - isn't the target "text" a named target as well?